targets/csharp/templates/PlayFabUtil.cs.ejs

```diff
@@ -49,6 +51,82 @@ namespace PlayFab
                 return null;
             return error.GenerateErrorReport();
         }
+
+#if NET45 || NETCOREAPP2_0
+        [ThreadStatic]
+        private static StringBuilder _sb;
+        /// <summary>
+        /// A threadsafe way to block and load a text file
+        /// 
+        /// Load a text file, and return the file as text.
+        /// Used for small (usually json) files.
+        /// </summary>
+        public static string ReadAllFileText(string filename)
+        {
+            if (!File.Exists(filename))
+            {
+                return string.Empty;
+            }
+                
+            if (_sb == null)
+            {
+                _sb = new StringBuilder();
+            }
+            _sb.Length = 0;
+
+            using (var fs = new FileStream(filename, FileMode.Open))
+            {
+                using (BinaryReader br = new BinaryReader(fs))
+                {
+                    while (br.BaseStream.Position != br.BaseStream.Length)
+                    {
+                        _sb.Append(br.ReadChar());
+                    }
+                }
+            }
+            return _sb.ToString();
+        }
+
+        internal static string GetLocalSettingsFileProperty(string propertyKey)
```

Comment:
Can this be private instead of internal? Or do we need it to be internal for testing purposes?

Reply:
It needs to be accessed by `PlayFabSettings.cs`

Reply:
OK. Should ReadAllFileText be private then?

Reply:
Unfortunately I believe it cannot be. 
C:\github\PlayFab\Fork\SDKGenerator\targets\unity-v2\Testing\Tests\Shared\TestTitleDataLoader.cs(49):                    testTitleDataContents = PlayFabUtil.ReadAllFileText(filename);

Reply:
OK. That makes sense. Did you add the test that calls ReadAllFileText? 

What I'm concerned about is that we have two methods here, both of which are called from outside the class, but both have different access modifiers. I don't understand why they are not the same (either both public or both internal). If one is only called by test code, but the other is called by customer code, then I could understand them being different as in that case the one called by customer code would be public and the test one would (or at least could) be internal (assuming InternalsVisibleTo attribute exists somewhere).

Reply:
Actually I apologize; this method is not being used by the C# SDK. That search query returned a result for a **unity** SDK test call to `ReadAllFileText`. This method can indeed be private as it is being used by code in `PlayFabUtil.cs` only.

The reason for all this confusion is the fact that the SDK team created this method for the C# SDK before we even started working on local-debugging. However, during our work on it, they decided to remove the method as it was no longer needed since they changed their approach for getting test title data on the C# SDK due to all the complexities of doing File IO when targetting 4 frameworks that not all support doing it (portable4.5).

It is due to this fact that the earlier version of the code you saw in this looked completely different in terms of coding style, such as the lack of `using` and `{` which is a pattern I've gotten used to while working here.

In short, we decided to bring this method back as we **need** it for our work but at the same time decided to `#if` define it because we don't expect people to want to run local-debugging on a portable 4.5 device.

So yes, this method can be private, and it will be in my commit shortly.

Reply:
Cool. Thanks for reminding me how we got here.